buildSrc/src/main/kotlin/TargetsConfig.kt

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+@file:Suppress("UNUSED_VARIABLE")
+
+import org.gradle.api.*
+import org.gradle.kotlin.dsl.*
+import org.jetbrains.kotlin.gradle.targets.native.tasks.*
+
+fun Project.configureTargets() {
+    val coroutines_version: String by extra
+    val kotlin_version: String by extra
+
+    val files = project.projectDir.listFiles() ?: emptyArray()
+
+    val hasCommon = files.any { it.name == "common" }
+    val hasJvmAndNix = hasCommon || files.any { it.name == "jvmAndNix" }
+    val hasPosix = hasCommon || files.any { it.name == "posix" }
+    val hasDesktop = hasPosix || files.any { it.name == "desktop" }
+    val hasNix = hasPosix || files.any { it.name == "nix" }
+    val hasDarwin = hasNix || files.any { it.name == "darwin" }
+    val hasJs = hasCommon || files.any { it.name == "js" }
+    val hasJvm = hasCommon || files.any { it.name == "jvm" }
+
+    kotlin {
+        if (hasJvm) {
+            jvm()
+            configureJvm()
```

Comment:
`configureJvm` also installs `jvm` target. for consistency may be it's better to leave target installing in one place?

Reply:
Yep, it should work. Fixed